web/packages/teleport/src/Bots/Details/BotDetails.tsx

```diff
@@ -74,7 +75,12 @@ export function BotDetails() {
   });
 
   const handleBackPress = () => {
-    history.goBack();
+    // If location.key is unset, or 'default', this is the first history entry in-app in the session.
+    if (!location.key || location.key === 'default') {
+      history.push(cfg.getBotsRoute());
+    } else {
+      history.goBack();
+    }
```

Comment:
i thought this meant, if there were no history, then push a history, otherwise go back to previous route in the app, but this seems to always default back to the bots list route

also i was just curious, i saw a similar code earlier and it was wrapped in useCallback but not here?